spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/HqlQueryTransformer.java

```diff
@@ -235,14 +235,28 @@ public List<JpaQueryParsingToken> visitFromRoot(HqlParser.FromRootContext ctx) {
 
 		if (ctx.entityName() != null) {
 
-			tokens.addAll(visit(ctx.entityName()));
+			List<JpaQueryParsingToken> entityName = visit(ctx.entityName());
+
+			tokens.addAll(entityName);
 
 			if (ctx.variable() != null) {
 				tokens.addAll(visit(ctx.variable()));
 
 				if (this.alias == null && !isSubquery(ctx)) {
 					this.alias = tokens.get(tokens.size() - 1).getToken();
 				}
+			} else {
+
+				if (countQuery) {
+					tokens.add(TOKEN_AS);
+
+					String tempAlias = entityName.get(0).getToken().substring(0, 1).toLowerCase();
```

Comment:
`tempAlias` isn't a great name. I'm concerned also whether the first character is a good fit for an alias, if the entity name starts with a non-character letter. Do we also have a test case for `"FROM BookError b …"`?